torch/_inductor/codegen/cuda/cutlass_lib_extensions/gemm_operation_extensions.py

```diff
@@ -0,0 +1,181 @@
+    ${element_c}, ${layout_c}, ${align_c},
+    ${element_d}, ${layout_d}, ${align_d},
+    EpilogueScheduleType,
+    ${operation_name}_epilogue_functor
+  >::CollectiveOp;
+
+using ${operation_name}_mainloop =
+  typename cutlass::gemm::collective::CollectiveBuilder<
+    ${arch}, ${opcode_class},
+    ${element_a}, ${layout_a}, ${align_a},
+    ${element_b}, ${layout_b}, ${align_b},
+    ${element_accumulator},
+    cute::Shape<cute::_${tile_shape_m}, cute::_${tile_shape_n}, cute::_${tile_shape_k}>,
+    cute::Shape<cute::_${cluster_m},cute::_${cluster_n},cute::_${cluster_k}>,
+    ${stages},
+  ${kernel_schedule}
+  >::CollectiveOp;
+
+// Gemm operator ${operation_name}
+using ${operation_name}_base = cutlass::gemm::kernel::GemmUniversal<
+    cute::Shape<int,int,int,int>,
+    ${operation_name}_mainloop,
+    ${operation_name}_epilogue,
+    ${tile_scheduler}>;
+
+// Define named type
+struct ${operation_name} :
+  public ${operation_name}_base { };
+
+"""
+
+    #
+    def instance_template(self):
+        return """
+${compile_guard_start}
+  using GemmKernel = cutlass::gemm::device::GemmUniversalAdapter<${operation_name}>;
+  manifest.append(
+    new ${gemm_kind}<GemmKernel>("${operation_name}"));
+${compile_guard_end}
+"""
+
+    #
+    def emit(self, operation):
+        tile_shape = operation.tile_description.tile_shape
+        warp_count = operation.tile_description.warp_count
+        # stage count set to zero indicates builder automatic stage selection
+        if operation.tile_description.stages > 0:
+            stage_count_string = f"cutlass::gemm::collective::StageCount<{str(operation.tile_description.stages)}>"
+        else:
+            stage_count_string = f"cutlass::gemm::collective::StageCountAutoCarveout<sizeof(typename {str(operation.procedural_name())}_epilogue::SharedStorage)>"  # noqa: B950
+        warp_shape = [tile_shape[idx] // warp_count[idx] for idx in range(3)]
+
+        instance_layout_A, instance_layout_B, instance_layout_C, instance_layout_D = (
+            operation.A.layout,
+            operation.B.layout,
+            operation.C.layout,
+            operation.D.layout,
+        )
+
+        # 3.0 profiler integration only supports trivial epilogues for now
+        epilogue_vector_length = 1
+
+        # Support built-in epilogue functors or user-defined functions
+        if isinstance(operation.epilogue_functor, enum.Enum):
+            values = {
+                "epilogue_vector_length": str(epilogue_vector_length),
+                "element_epilogue": str(DataTypeTag[operation.element_epilogue]),  # type: ignore[name-defined]
+                "epilogue_functor": EpilogueFunctorTag[operation.epilogue_functor],  # type: ignore[name-defined]
+            }
+            epilogue_functor = SubstituteTemplate(  # type: ignore[name-defined]
+                self.builtin_epilogue_functor_template, values
+            )
+
+        elif callable(operation.epilogue_functor):
```

Comment:
These needed to be adapted a bit as well, so we can just pass in the epilogue_functor string instead of building it via callback or such.